addons/mail/models/mail_message.py

```diff
@@ -1153,6 +1184,32 @@ def _get_store_attachment_fields(self, target):
             return self.env["ir.attachment"]._get_store_ownership_fields()
         return []
 
+    def _get_store_linked_messages_fields(self):
+        """Add the messages that are referenced by the current message's body to the given store."""
+        record_by_message = self.linked_message_ids._record_by_message()
+        # store.target is not checked: the current user might broadcast data about messages they are
+        # allowed to see to other users, which is considered acceptable as only the existence of the
+        # message and the record name are returned.
+        return [
+            Store.Many(
+                "linked_message_ids",
```

Comment:
Comment could be reworded slightly because this is hard to read.
I would remove the part after `,` and also build sentence like other comment in this commit, "# Remove any potential sudo from the env as [...]", which is more straight-forward in intention of action and then justifies, rather than state hypothetical problem. For example something like this:

_"return data of message that is allowed for any user, as the store.target is not checked and can be broadcast to anyone"_